dbt/adapters/spark/connections.py

```diff
@@ -32,17 +32,25 @@ class SparkCredentials(Credentials):
     host: str
     method: SparkConnectionMethod
     schema: str
-    cluster: Optional[str]
-    token: Optional[str]
-    user: Optional[str]
     database: Optional[str]
+    cluster: Optional[str] = None
+    token: Optional[str] = None
+    user: Optional[str] = None
     port: int = 443
     organization: str = '0'
     connect_retries: int = 0
     connect_timeout: int = 10
 
     def __post_init__(self):
         # spark classifies database and schema as the same thing
+        if (
+            self.database is not None and
+            self.database != self.schema
+        ):
+            raise dbt.exceptions.RuntimeException(
+                f'In credentials: got database={self.database} but '
+                f'schema={self.schema} - on spark, both most be the same value'
```

Comment:
Based on when this error appears, I think we can massage the wording a bit:
```
Running with dbt=0.17.0-rc1
Encountered an error while reading profiles:
  ERROR Runtime Error
  Credentials in profile "garage-spark", target "prod" invalid: Runtime Error
    In credentials: got database=other but schema=dbt - on spark, both most be the same value
Defined profiles:
```

```suggestion
                f'    schema: {self.schema} \n'
                f'    database: {self.database} \n'
                f'On Spark, database must be omitted or have the same value as'
                f' schema.'
```